src/main/java/hudson/tasks/junit/CaseResult.java

```diff
@@ -264,7 +275,11 @@ private static String getSkippedMessage(Element testCase) {
     }
 
     public String getDisplayName() {
-        return TestNameTransformer.getTransformedName(testName);
+        StringBuilder transformedName = new StringBuilder(TestNameTransformer.getTransformedName(testName));
+        if (this.archiveId != null) {
+            transformedName.insert(0, "[" + this.archiveId + "]");
+        }
+        return transformedName.toString();
```

Comment:
NIT: Seems like a funky way of appending the strings to each other. Isn't this what you're effectively doing...

```java
        StringBuilder transformedName = new StringBuilder();
        if (this.archiveId != null) {
            transformedName.append("[" + this.archiveId + "]");
        }
        transformedName.append(TestNameTransformer.getTransformedName(testName));
```

Seems a little easier to read to me, but obviously subjective (hence NIT).